src/Microsoft.Health.Dicom.Core/Features/CustomTag/IndexTag.cs

```diff
@@ -13,15 +13,15 @@ namespace Microsoft.Health.Dicom.Core.Features.CustomTag
     /// </summary>
     public class IndexTag
     {
-        public IndexTag(DicomTag tag, DicomVR vr, CustomTagLevel level, bool isCustomTag)
+        public IndexTag(DicomTag tag, DicomVR vr, CustomTagLevel level, CustomTagStoreEntry customTagStoreEntry)
         {
             EnsureArg.IsNotNull(tag, nameof(tag));
             EnsureArg.IsNotNull(vr, nameof(vr));
 
             Tag = tag;
             VR = vr;
-            IsCustomTag = isCustomTag;
             Level = level;
```

Comment:
You probably need 2 different constructors, one for standard and one for custom. Not sure you still need to take CustomTagLevel

Reply:
Still keep existing constructor as simple as now, created another 2 static method for converting.

Reply:
It still has the same problem. 
CustomTagLevel level. Why do callers need to pass this? When you can get this from the customTagStoreEntry

Adding 2 constructors helps to check for nulls also

Reply:
will update.